File: 프로그래머스/1/12982. 예산/예산.py
def solution(d, budget):
    
    #지원한 부서랑 사용예산 변수
    a = 0
    result = 0
    
    d.sort()
    #인덱스 순회, 신청 금액
    for i in range(0, len(d)):
        if result + d[i] <= budget:
        #사용한 예산이랑 신청금액 더하고 이게 예산 넘지않으면 지원
            result += d[i]
        #예산 사용
            a += 1
        else:
            break
    
    return a

File: 프로그래머스/1/12982. 예산/예산.java
import java.util.Arrays;

class Solution {
    public int solution(int[] d, int budget) {
        int answer = 0;
        int sum = 0;
        
        // 배열을 오름차순으로 정렬
        Arrays.sort(d);
        
        // 정렬된 배열을 순회하면서 예산을 분배
        for (int i = 0; i < d.length; i++) {
            sum += d[i];
            
            // 누적 합이 예산을 초과하면 반복문 종료
            if (sum > budget) {
                break;
            }
            
            answer++;
        }
        
        return answer;
    }
}
